### ISSUE TITLE
tabAutocomplete doesn't work correct with Open WebUI / Ollama since 0.9.224 / 0.8.56

### ISSUE THREAD

**Issue by vills** [continuedev/continue#2954]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://docs.continue.dev/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS: MacOS 15.0.1
- Continue version: 0.9.224+
- IDE version: VSCode
- Model: deepseeker-code-v2
- config.json:
  
  "tabAutocompleteModel": {
    "title": "DeepSeeker Coder v2 16b",
    "provider": "ollama",
    "model": "deepseek-coder-v2:16b-lite-instruct-q6_K",
    "apiBase": "https://OPENWEBUI/ollama",
    "apiKey": "open-webui-token",
    "contextLength": 8096,
    "completionOptions": {
      "temperature": 0.1,
      "topP": 0.95,
      "maxTokens": 600
    }
  },
```

### Description

I have a setup of autocompletitions with open-webui involved. Open-webui serves as a proxy to Ollama. It was working previously fine, but with latest versions of continue.dev plugin instead of autocomplete i get explanations of what my code does.

I checked previous versions and seems like problem arrive starting from 0.9.224 pre-release and 0.8.56 versions. Previous versions works as expected.

I believe it is a bug. But maybe you changed something in configs i need change too?

Thanks.

**Comment by sestinj** [comment]
@vills my guess is that this is caused by using the "ollama" provider for a non-Ollama API. The thing we did specifically lately was to let Ollama handle FIM formatting for us, but when going through OpenWebUI my guess is that the FIM endpoint doesn't exist or is routed to chat.

For my own understanding (you're not the only one to have this setup, and I haven't looked into it yet): what is the benefit of routing through OpenWebUI instead of directly to Ollama? If I had to point you to a solution, it would definitely be to just go direct, but I figure there's a reason you can't.

**Comment by vills** [comment]
Thanks for sharing your idea. I'll check it in a few days.

I use ollama behind open-webui as it provides me with (obviously) UI and authorization/authentication (i use remote server with GPU that i share with my friends). Plus it can route to different payed platforms. It's kind of very simple to setup and get this features.

**Comment by uganson** [comment]
I am also experiencing the same issue, since 0.8.56 the tab autocompletions show a description of the code before the cursor rather than a code suggestion, where prior versions work as expected. I am connecting directly to an ollama server running locally.

**Comment by arnesund** [comment]
Also hitting this issue. The big benefit of Open WebUI in this mix is that it handles authentication with integration with SSO providers and has API keys per user. In our setup that enable all developers in the company to use Open WebUI as the publicly available LLM proxy, reachable from their laptops, while Ollama itself is not reachable from the Internet.

**Comment by vills** [comment]
I can confirm accessing remote Ollama instance works well. Looks like open-webui's proxy do something with with original requests. I'll try to dig deeper..

**Comment by tomasz-stefaniak** [comment]
@vills you may want to take a look at the `CompletionProvider` code, specifically the [provideInlineCompletionItems](https://github.com/continuedev/continue/blob/tomasz/fix-github-issue-provider/core/autocomplete/CompletionProvider.ts#L136) function. [streamCompletionWithFilters](https://github.com/continuedev/continue/blob/tomasz/fix-github-issue-provider/core/autocomplete/CompletionProvider.ts#L208) might also be of interest.

**Comment by vills** [comment]
I made some tests and responses from direct Ollama connection and Open WebUI proxied are differ - https://github.com/open-webui/open-webui/discussions/6720#discussioncomment-11335817.

I think Open WebUI do not proxies "suffix" parameter to Ollama and that's the issue with autocomplete.

**Comment by vills** [comment]
@sestinj one more question. I see in codebase you detect "use FIM or not" by searching for ".Suffix" parameter in Ollama's model template. Is there a manual way to revert this behavior while bug isn't fixed in Open WebUI?

**Comment by vills** [comment]
FYI, fixed in open-webui v0.4.3 release.

**Comment by tomasz-stefaniak** [comment]
Great, thank you for taking care of this @vills!

Does this mean this issue can be closed now?
